xdsl/dialects/arm_neon.py

```diff
@@ -236,6 +236,59 @@ def assembly_line_args(self):
         )
 
 
+@irdl_op_definition
+class DSSFMulVecOp(ARMInstruction):
+    """
+    Floating-point Multiply (vector). This instruction multiplies corresponding
+    floating-point values in the vectors in the two source NEON registers,
+    places the result in a vector, and writes the vector to the destination.
+
+    See external [documentation](https://developer.arm.com/documentation/100069/0606/SIMD-Vector-Instructions/FMUL--vector-).
+    """
+
+    name = "arm_neon.dss.fmulvec"
+    d = result_def(NEONRegisterType)
+    s1 = operand_def(NEONRegisterType)
+    s2 = operand_def(NEONRegisterType)
+    arrangement = prop_def(NeonArrangementAttr)
+
+    assembly_format = "$s1 `,` $s2 $arrangement attr-dict `:` `(` type($s1) `,` type($s2) `)` `->` type($d)"
```

Comment:
```suggestion
    assembly_format = "$s1 `,` $s2 $arrangement attr-dict `:` functional-type(operands, $d)"
```
This should also work, though don't mind what is here now.